### PR TITLE
Fix crash on redirection when path contains unicode characters

### PR DESCRIPTION
**Pull request by leplatrem**

### CODE REVIEW

**Comment by leplatrem** [comment]
You know what, I'm done with this. Life is too beautiful to ruin it with crazy sh$¤ like UnicodeDecodeError and bytes-like object required.

**Comment by Natim** [comment]
It is not your fault @leplatrem this thing is shitty...

File: tests/core/test_views_errors.py [hunk 1]
```diff
@@ -183,6 +183,12 @@ def test_redirection_does_not_allow_crlf(self):
     def test_redirection_does_not_allow_control_characters(self):
         self.app.get('/9l2j7%0A21m2n', status=404)
 
+    def test_redirection_allows_unicode_characters(self):
+        self.app.get('/crlftest%E5%98%8AXTest:crlftest/', status=307)
```
Comment (glasserc) [inline tests/core/test_views_errors.py]: Maybe add a comment showing the character that this represents? `嘊`

**Comment by glasserc** [comment]
🎆 ❇️ 💥

**Comment by leplatrem** [comment]
I was wrong about webob, at least regarding `request.path` (some other issues exist with parameters like https://github.com/Pylons/webob/issues/161).

The `request.matchdict` (Pyramid url dispatch) has unicode, whereas `request.path` hasn't.

Hence the great simplification.

**Pull request merged**